listenbrainz_spark/recommendations/train_models.py

```diff
@@ -154,6 +156,9 @@ def main():
         current_app.logger.error(str(err), exc_info=True)
         sys.exit(-1)
 
+    # Add checkpoint dir to break and save RDD lineage.
+    listenbrainz_spark.context.setCheckpointDir(config.HDFS_CLUSTER_URI + path.CHECKPOINT_DIR)
```

Comment:
Does Spark create the dir if it doesn't exist? If not, what creates it?

Reply:
Yes, Spark creates the directory if it does not exist.